commercial/app/model/commercial/books/BookFinder.scala

```diff
@@ -113,3 +120,46 @@ object MagentoService extends Logging {
     }
   }
 }
+
+
+object BookCache extends Logging with MemcachedCodecs {
+
+  private implicit lazy val executionContext = memcachedExecutionContext
+  private implicit val stringCodec = StringBinaryCodec
+
+  private lazy val maybeCache: Option[Memcached] = {
+    val maybeHost = Configuration.memcached.host
+    for (host <- maybeHost) yield {
+      val config = MemcachedConfiguration(
+        addresses = host,
+        keysPrefix = Some("model.commercial.book")
+      )
+      Memcached(config, memcachedExecutionContext)
+    }
+  }
+
+  private def withCache[T](action: Memcached => Future[T]): Future[T] = {
+    maybeCache map action getOrElse {
+      log.warn("Cache not configured")
+      Future.failed(CacheNotConfiguredException)
+    }
+  }
+
+  def get(isbn: String): Future[Option[Book]] = withCache { cache =>
+    val bookJson = cache.get[String](isbn)
+    bookJson onFailure {
+      case NonFatal(e) => log.error(s"Fetching book from cache failed: ${e.getMessage}")
```

Comment:
@rich-nguyen: I don't think you can catch fatal exceptions in any useful way.  They're really not exceptions at all but errors like stackoverflows, outofmemory etc.